internal/persistence/sql/uuid_mapping.go

```diff
@@ -24,17 +23,32 @@ func (UUIDMapping) TableName() string {
 	return "keto_uuid_mappings"
 }
 
-func (p *Persister) AddUUIDMapping(ctx context.Context, id uuid.UUID, representation string) error {
-	m := &UUIDMapping{
-		ID:                   id,
-		StringRepresentation: representation,
-	}
+func (p *Persister) ToUUID(ctx context.Context, text string) (uuid.UUID, error) {
+	id := uuid.NewV5(p.NetworkID(ctx), text)
```

Comment:
One more thing I realized: subjects are equal by string comparison, but objects are namespace bound. This means that the objects in the tuples `groups:root#member@foo` and `files:root#access@foo` are not referring to the same object, but the subjects are the same in those tuples.
Therefore I think we have to add the namespace to the UUID step as well. Not sure how though, because if we just string-append the namespace and object, it can collide with a subject that is that string. Additionally namespace names can change, but the namespace numeric ID is constant.

What do you think of
```suggestion
	id := uuid.NewV5(uuid.NewV5(p.NetworkID(ctx), strconv.Itoa(namespace.ID)), text)`
```
for objects, and yours for subjects?

Reply:
Is that really an issue? I assumed that the objects would be considered namespace-bound elsewhere, so even if the same UUID appears in multiple namespaces (as now arbitrary strings would), this would not be a problem.

In summary, I would rather take the namespace into consideration for object equivalence rather than relying on them being mapped to different UUIDs.

Reply:
True, good point :+1:

Reply:
OK nice, I think we have everything resolved then :)